### ISSUE TITLE
Allow configuring `docs_dir` in `mkdocs serve`

### ISSUE THREAD

**Issue by dev-matan-tsuberi** [mkdocs/mkdocs#1399]

**Comment by waylan** [comment]
I'm not sure what value this has. A `mkdocs.yml` config file should be unique to a specific `docs_dir`, so I can't imagine a scenario where you would need to point to a different `docs_dir`. And you can already specify a config file wit the `--config-file` or `-f` flag. The only reason it is configurable at all is because some users may want their directory to have a different name than the default. We do not intend use of the same config for multiple different sets of documentation.

And when #1376 gets finalized all paths in a config will be relative to the location of the config file. In that case, providing a `docs_dir` different than what is in the config makes even less sense.

To sum up, any given config file is unique to a specific docs_dir. Being able to override that value from the CLI breaks that.

**Comment by dev-matan-tsuberi** [comment]
I think you should always allow the option. Maybe include a warning when using this flag explaining what you said.

I'm asking for this feature because it's required for an automation task I'm doing and otherwise I would need to do an ugly hack to make it work.

**Comment by waylan** [comment]
> it's required for an automation task I'm doing 

And I assume the automation task is run from a  different dir than the location of the config file. Of course currently that is broken (see #543), but when all paths in the config are relative to the config (#1376), that won't be a problem anymore. This and that change seem to be mutually exclusive to me. As we will be making that change, I see no value in this. In other words, we already have a different fix in the works. Your going to need to provide a more specific use case to demonstrate that #1376 doesn't solve your problem.

**Comment by dev-matan-tsuberi** [comment]
No, my script runs in the same dir as `mkdocs.yml`.

In my task, before running `mkdocs build` or `mkdocs deploy`, a compile step is needed to generate `.md` files from doc-strings in the source files. This compilation step takes files from `src` merges it with files from `docs` and outputs the result in `.docs_build`. So we set `docs_dir` to `.docs_build`

This is all fine until you want to `mkdocs serve` the website locally while editing files in `docs`, you don't want to have to re-compile all the files in order to see the changes.
So, I need `mkdocs build/deploy` to have `docs_dir: .docs_build` and `mkdocs serve` to have `docs_dir: docs`. Currently my script edits the `mkdocs.yml` file before running the command, which is very ugly.

So this is my use case, it may be narrow, but again, allowing the option doesn't hurt and you can make sure the user is aware of the problems you mention by including a warning message.

**Comment by waylan** [comment]
That seems like a reasonable use case. In fact, there are a number of issues which arise from people combining MkDocs with other tools. I think we probably need to take a holistic approach to all of them. Whether that means we make the docs_dir configurable from the CLI or we come up with some other solution I'm not sure.

**Comment by froschdesign** [comment]
@dev-matan-tsuberi 
> Currently my script edits the mkdocs.yml file before running the command, which is very ugly.

Why is this ugly?
The same solution is used for all components of Zend Framework.

See: https://github.com/zendframework/zf-mkdoc-theme/blob/296c5ac97080a7419c541d333295cecef1d27606/build.sh#L45-L55

**Comment by waylan** [comment]
As of version 1.0 we now have an `on_files` plugin event which can be used to add files which are not in the `docs_dir`. Therefore, the "proper" way to include external files, is not to copy them to the `docs_dir`, but to use a plugin to list them in the files collection. In other words, the stated reason for this request is now obsolete.
